url2f44efcfa6_wangyudong_sohutw_java_sdk_2_0_tgz-pJ8-com_sohu_t_api_sdk_json_TestJ8/src/com/sohu/t/api/sdk/data/User.java

```diff
@@ -0,0 +1,359 @@
+package com.sohu.t.api.sdk.data;
+
+import com.sohu.t.api.sdk.TBlogException;
+import com.sohu.t.api.sdk.TBlogResponse;
+import com.sohu.t.api.sdk.http.Response;
+import com.sohu.t.api.sdk.json.JSONArray;
+import com.sohu.t.api.sdk.json.JSONException;
+import com.sohu.t.api.sdk.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * User: chenxiaojian
+ * Date: 13-1-8
+ * Time: 下午12:31
+ */
+public class User extends TBlogResponse {
+
+    private String id = "";
+    private String screen_name = "";
+    private String name = "";
+    private String location = "";
+    private String description = "";
+    private String url = "";
+    private String gender = null;
+    private String profile_image_url = "";
+    private boolean p_rotected = false;
+    private int followers_count = 0;
+    private String profile_background_color = "";
+    private String profile_text_color = "";
+    private String profile_link_color = "";
+    private String profile_sidebar_fill_color = "";
+    private String profile_sidebar_border_color = "";
+    private int friends_count = 0;
+    private String created_at = "";
+    private int favourites_count = 0;
+    private String utc_offset = "";
+    private String time_zone = "";
+    private String profile_background_image_url = "";
+    private String notifications = "";
+    private boolean geo_enabled = false;
+    private int statuses_count = 0;
+    private boolean following = false;
+    private boolean verified = false;
+    private String lang = "zh_cn";
+    private boolean contributors_enabled = false;
+    private String passport = null;
+    private String password = null;
+    private RetweetedStatus status;
+
+    public User(Response res) throws TBlogException {
+        init(res.asJSONObject());
+    }
+
+    public User(JSONObject json) throws TBlogException {
+        init(json);
+    }
+
+    public static List<User> constructUsers(JSONArray jsonArray) throws TBlogException {
+        try {
+            int size = jsonArray.length();
+            List<User> users = new ArrayList<User>(size);
+            for (int i = 0; i < size; i++) {
+                users.add(new User(jsonArray.getJSONObject(i)));
+            }
+            return users;
+        } catch (JSONException jsone) {
+            throw new TBlogException(jsone);
+        } catch (TBlogException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)